### PR TITLE
Steem 0.18.103

### PR DESCRIPTION
**Pull request by holgern**
https://github.com/steemit/steem-python

### CODE REVIEW

File: recipes/steem/bld.bat [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pip install scrypt==0.8.0
```
Comment (CJ-Wright) [inline recipes/steem/bld.bat]: please install these via conda-forge

File: recipes/steem/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    - autopep8
+    - yapf
+    - sphinxcontrib
+    - recommonmark
+    - pypandoc
+    - pep8
+    - pkgconfig                    # [unix]
+    # For sha256
+    - openssl 1.0.*                # [unix]
+    - setuptools
```
Comment (CJ-Wright) [inline recipes/steem/meta.yaml]: Are both `setuptools` and `pip` required?

File: recipes/steem/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    - ecdsa
+    - pycrypto
+    - toolz
+    - urllib3
+    - w3lib
+    - funcy
+    - langdetect
+    - prettytable
+    - voluptuous
+    - sphinx
```
Comment (CJ-Wright) [inline recipes/steem/meta.yaml]: Why are doc building tools build requirements?

File: recipes/steem/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    - autopep8
```
Comment (CJ-Wright) [inline recipes/steem/meta.yaml]: Is this a build requirement?

File: recipes/steem/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    - urllib3
+    - w3lib
+    - funcy
+    - langdetect
+    - prettytable
+    - voluptuous
+
+test:
+  imports:
+    - steem
```
Comment (CJ-Wright) [inline recipes/steem/meta.yaml]: Could you please add more import tests?

Comment (holgern) [reply recipes/steem/meta.yaml]: I will add all possible combinations

File: recipes/steem/meta.yaml (hunk 2 above)
Comment (holgern) [reply recipes/steem/meta.yaml]: I will try to remove one

File: recipes/steem/meta.yaml (hunk 3 above)
Comment (holgern) [reply recipes/steem/meta.yaml]: I will try to remove sphinx

File: recipes/steem/meta.yaml (hunk 4 above)
Comment (holgern) [reply recipes/steem/meta.yaml]: I will try to remove

**Comment by holgern** [comment]
I use now the scrypt build from conda-forge.

But there are three more packages from pip, which are necessary:
pip install sphinxcontrib-restbuilder
pip install sphinxcontrib-programoutput
pip install pytest-pylint

Should I add them to conda-forge first?

**Comment by holgern** [comment]
Ok , added https://github.com/conda-forge/staged-recipes/pull/5109, https://github.com/conda-forge/staged-recipes/pull/5107 and https://github.com/conda-forge/staged-recipes/pull/5106

**Comment by CJ-Wright** [comment]
Are these sphinx requirements for building the STEEM docs?

File: recipes/steem/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/07/7e/142fb6db1e14db39e658832b9d1b7f493d2e856ecd9c17f08490f85bf71e/steem-{{ version }}.tar.gz
+  sha256: {{ sha256|lower }}
+
+build:
+  number: 0
+  skip: True  # [py<36]
+  script: python setup.py install --single-version-externally-managed --record record.txt
```
Comment (CJ-Wright) [inline recipes/steem/meta.yaml]: You might try `python -m pip install --no-deps --ignore-installed .`
You will need to add pip to the deps, but then it should ignore all the stated deps.

Comment (CJ-Wright) [reply recipes/steem/meta.yaml]: This way you can manage the deps directly without `setuptools` complaining.

Comment (holgern) [reply recipes/steem/meta.yaml]: Thanks, I will try

**Comment by holgern** [comment]
I have to wait for https://github.com/conda-forge/staged-recipes/pull/5107

**Comment by holgern** [comment]
Now i have to wait for https://github.com/conda-forge/staged-recipes/pull/5109

**Comment by holgern** [comment]
I seperated build and run requirements.
The package is ready for a second review:).

File: recipes/steem/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    - steembase.bip38
+    - steembase.chains
+    - steembase.exceptions
+    - steembase.http_client
+    - steembase.operationids
+    - steembase.operations
+    - steembase.storage
+    - steembase.transactions
+    - steembase.types
+  commands:
```
Comment (CJ-Wright) [inline recipes/steem/meta.yaml]: You might consider testing the CLI as well (just make sure that running the help doesn't blow things up)

File: recipes/steem/meta.yaml [hunk 8]
```diff
@@ -80,9 +80,11 @@ test:
     - steembase.storage
     - steembase.transactions
     - steembase.types
+
   commands:
     - conda inspect linkages -p $PREFIX steem  # [not win]
     - conda inspect objects -p $PREFIX steem   # [osx]
+    - python steempy --help
```
Comment (CJ-Wright) [inline recipes/steem/meta.yaml]: Does this require a call to python? https://github.com/steemit/steem-python#cli-tools-bundled

File: recipes/steem/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/07/7e/142fb6db1e14db39e658832b9d1b7f493d2e856ecd9c17f08490f85bf71e/steem-{{ version }}.tar.gz
+  sha256: {{ sha256|lower }}
+
+build:
+  number: 0
+  skip: True  # [py<36]
+  script: python -m pip install --no-deps --ignore-installed .
+
```
Comment (CJ-Wright) [inline recipes/steem/meta.yaml]: You need to declare entry points (cli stuff) in the build https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#python-entry-points

**Comment by holgern** [comment]
I have to wait for https://github.com/conda-forge/staged-recipes/pull/5129

File: recipes/steem/meta.yaml (hunk 9 above)
Comment (holgern) [reply recipes/steem/meta.yaml]: Thanks for the tip.
I added them.

**Comment by CJ-Wright** [comment]
Do these build on your local machine?

**Comment by holgern** [comment]
Problem is that packages are missing. I'm waiting for https://github.com/conda-forge/staged-recipes/pull/5138

**Comment by CJ-Wright** [comment]
I guess my point was that you could keep testing for the next req on your local rather than pinging it off the CI.

**Comment by holgern** [comment]
Sorry, you are right. I'm doing this now.

**Comment by holgern** [comment]
There is a bug in the toml package. It is not found by pkg_resources.require("toml")
I created a pull request https://github.com/conda-forge/toml-feedstock/pull/3, in order to fix this.

**Comment by holgern** [comment]
The package can be merged, It does not need toml, with the newest patch. Pipfile and toml are only used within the Docker build, so I removed it. It is also removed in the newest git commit of steem-python.

**Pull request merged**